pages/api/lnurlp/[username]/index.js

```diff
@@ -1,20 +1,43 @@
+import { randomBytes } from 'crypto'
 import { getPublicKey } from 'nostr'
 import models from '../../../../api/models'
 import { lnurlPayMetadataString } from '../../../../lib/lnurl'
 import { LNURLP_COMMENT_MAX_LENGTH } from '../../../../lib/constants'
 
+const generateK1 = () => randomBytes(32).toString('hex')
+
+export const k1Cache = new Map()
+
 export default async ({ query: { username } }, res) => {
   const user = await models.user.findUnique({ where: { name: username } })
   if (!user) {
     return res.status(400).json({ status: 'ERROR', reason: `user @${username} does not exist` })
   }
 
+  // Generate a random k1, cache it with the requested username for validation upon invoice request
+  const k1 = generateK1()
+  k1Cache.set(k1, username)
+  // Invalidate the k1 after 10 minutes, if unused
+  setTimeout(() => {
+    k1Cache.delete(k1)
+  }, 1000 * 60 * 10)
+
```

Comment:
I went with an in-memory cache because it felt like overkill to store this in the DB. However, I am not certain of SN's deployment architecture (i.e. are there several server instances behind a load balancer?), so if we need to push the k1 cache to the DB to share across instances, I can make those changes. Just let me know.

Reply:
Yes we have serveral servers behind a load balancer

Reply:
OK, I'll move it to the DB, then. Thank you for the confirmation.